src/combine.cpp

```diff
@@ -863,6 +863,10 @@ void do_symbol_check(chunk_t *prev, chunk_t *pc, chunk_t *next)
       {
          set_chunk_type(pc, (prev->type == CT_ANGLE_CLOSE) ? CT_PTR_TYPE : CT_DEREF);
       }
+      if ((cpd.lang_flags & LANG_CPP) && (pc->type == CT_CARET) && (prev->type == CT_ANGLE_CLOSE))
```

Comment:
Note sure if this is correct.  You are checking for ">^", but I don't see that combo in the new tests.
Did you mean to check for "^>"?